src/Microsoft.AspNetCore.Razor.Language/RazorCodeDocumentExtensions.cs

```diff
@@ -3,13 +3,61 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using Microsoft.AspNetCore.Razor.Language.Intermediate;
 
 namespace Microsoft.AspNetCore.Razor.Language
 {
     public static class RazorCodeDocumentExtensions
     {
-        private static object TagHelperPrefixKey = new object();
+        // Internal for testing
+        internal static readonly string IdentifierKey = "identifier";
+        internal static readonly string ImportIdentifiersKey = "imports-identifiers";
+
+        public static string GetIdentifier(this RazorCodeDocument document)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            return document.Items[IdentifierKey] as string;
+        }
+
+        public static void SetIdentifier(this RazorCodeDocument document, string identifier)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            document.Items[IdentifierKey] = identifier;
+        }
+
+        public static IReadOnlyList<string> GetImportIdentifiers(this RazorCodeDocument document)
```

Comment:
This goes away once @ajaybhargavb moves the `GetIdentifier` to source document yes?